### PR TITLE
Infer api refactoring

### PR DESCRIPTION
**Pull request by quic-rishinr**
- Refactored infer module and removed redundant code
- Add methods for constructing qpc dir name and generating tokenizer

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "cSpell.words": [
+        "msgpack"
```
Comment (vbaddi) [inline .vscode/settings.json:3]: Do we need this file?

File: QEfficient/cloud/infer.py [hunk 2]
```diff
@@ -30,6 +31,56 @@
...
+    - hf_token (Optional[str]): The Hugging Face token for authentication.
+
+    Returns:
+    - AutoTokenizer: The tokenizer for the downloaded model.
+    - str: Downloaded model path
+    """
+    if hf_token is not None:
+        login(hf_token)
+    
+    model_hf_path = hf_download(
```
Comment (vbaddi) [inline QEfficient/cloud/infer.py:75]: Does it make sense to download entire model checkpoints just to get `tokenizer`?

File: QEfficient/cloud/infer.py [hunk 3]
```diff
@@ -49,117 +100,79 @@ def main(
...
-    assert (
-        qpc_dir_path == generated_qpc_path
-    ), f"QPC files were generated at an unusual location, expected {qpc_dir_path}; got {generated_qpc_path}"
-    logger.info(f"Compiled qpc files can be found at : {generated_qpc_path}")
-
-    # Execute
+
+    if qpc_path_exists:
+        # Continue to execute with pre-compiled QPC
+        logger.info("Pre-compiled QPC found! Executing with given prompt.")
```
Comment (vbaddi) [inline QEfficient/cloud/infer.py:123]: `execute `api() is missing?

File: .vscode/settings.json (hunk 1 above)
Comment (quic-rishinr) [reply .vscode/settings.json:3]: No. It came by mistake. Will remove it.

File: QEfficient/cloud/infer.py (hunk 2 above)
Comment (quic-rishinr) [reply QEfficient/cloud/infer.py:75]: it would be better to move it to a separate method.

Comment (ochougul) [reply QEfficient/cloud/infer.py:75]: Hey @quic-rishinr  this loading only tokenizer is being handled in PR 31, please check

**Review by ochougul — CHANGES_REQUESTED** [review]
Please check PR 31 changes.

**Comment by quic-rishinr** [comment]
These changes will be added as part of https://github.com/quic/efficient-transformers/pull/31

**Pull request closed without merging**